gordon/router.py

```diff
@@ -142,13 +172,22 @@ def _get_next_phase(self, event_msg):
                 msg = f'Adding message "{event_msg}" to success channel.'
                 event_msg.append_to_history(msg, event_msg.phase)
                 await self.success_channel.put(event_msg)
+            elif old_phase == 'publish':
```

Comment:
(love the word penultimate)

Paging @descent3me to this convo since she & I have worked/talked about the phase route. As it stands, there are fixed phases that Gordon supports, although we want to eventually support multiple phases/paths, we're limiting it now for mvp. So yes as it stands, `publish` is designated as the final phase before cleaning up.

Reply:
@descent3me what happens if `next_phase` is `cleanup` and `old_phase` is `consume`? I wouldn't want to count that by accident; just limiting it to the full happy path.

Reply:
Aha, I see what you're trying to measure now. For the metric I was thinking of I would expect it to get emitted even if old_phase if 'consume', because it still shows the message was routed through successfully.

What you are describing sounds like a good metric to track, but it also sounds very implementation-specific. I would expect to track that in the plugin itself rather than the router (i.e. the GDNSPublisher emits a 'successfully-published' metric).